src.ts/transaction/transaction.ts

```diff
@@ -109,41 +109,46 @@ export interface TransactionLike<A = string> {
      *  The versioned hashes (see [[link-eip-4844]]).
      */
     blobVersionedHashes?: null | Array<string>;
+
+    /**
+     * Custom chain specific data.
+     */
+    customData?: any;
```

Comment:
you could make this a generic like 
```
type TransactionLike<A = string, CD = any>
customData: CD
```

then `type CeloTranactionLike = TransactionLike<string, {feeCurrency: string}>`